### PR TITLE
Addison Cole/Pushpa Agr Adagrams submission

### PR DESCRIPTION
**Pull request by addisoncole**
# Adagrams
Congratulations! You're submitting your assignment.

## Comprehension Questions

Feature | Feedback
--- | ---
What are the components that make up a method? | the method signature which is made up of the method name (used to invoke it), and its parameters. It is started with a 'def' or define clause and ends with 'end'. Inside of the method is a block of code of which arguments which are the values passed in through the method as parameters and which the block will process and return a result.
What are the advantages of using git when collaboratively working on one code base? | you can make commits and see the version history change in chronologincally. You can work from differnet locations or or using different machines, but still maintain a single code across work stations
What kind of relationship did you and your pair have with the unit tests? | we usually spent the first 8-10 minutes reading the problem silently on our own and then fashioned some form of pseudocode. We would then both describe our steps of achieving the goal to each other, and discuss why we might use one method over another, or discuss the merits of our decisions in writing the pseudocode. We would then choose who drove and who navigated and would work through the problem recoordinating code if something didnt work as planned. 
Does your code use any methods from the `Enumerable` mixin? If so, where and why was it helpful? |
What was one method you and your pair used to debug code? | We used slice to help maintain letter ratios in the drawing letter method, by deleting each one as it was used from the array.The array contained all the possible letters, individually. We made the array from an original hash containing all the letters and their ratios. This way it was nondestructive any time letters were drawn, and in the future a developer could easily change the ratios without having to go through the array and count each letter individually. 
What are two discussion points that you and your pair discussed when giving/receiving feedback from each other that you would be willing to share? | We had many conversations about how to structure our letter pool and we changed that structure probably 4 or 5 times. The first few times was because as we started writing the code we found better ways to structure it that would make the method shorter and keep the ratios correct throughout the letter drawing. We ended up also incorporating the letter pool  and initializing it as a hash and then iterating it into an array to do this. We also discussed several ways to wave 4 and went back and forth on the logic of it several times or the best way to store our words and scores.

### CODE REVIEW

**Comment by addisoncole** [comment]
Answer 5 should be answer 4. Answer 4 is:
We used a combination of pry and some puts statements

**Comment by tildeee** [comment]
# Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | you are free to edit your PR info and update the table of question/answers! but yes, got it :)
Both teammates contributed to the codebase | x
Small commits with meaningful commit messages | x
**Code Requirements** |
`draw_letters` method | x
Uses appropriate data structure to store the letter distribution | x
All tests for `draw_letters` pass | x
`uses_available_letters?` method | x
All tests for `uses_available_letters?` pass | x
`score_word` method | x
Uses appropriate data structure to store the letter scores | x
All tests for `score_word` pass | x
`highest_score_from` method | x
Appropriately handles edge cases for tie-breaking logic | x
All tests for `highest_score_from` pass | x
**Overall** |

Good job on this project, y'all!

The code meets all of the requirements and passes. Great work!

There are some places where the syntax is not what I would expect, and I'm making a few comments on that.

One thing I want to ask a question is: You all took out the question marks out of some method names. Turns out in ruby-- you can include `?`s in a method name! Adding one to the end of a method name, such as `uses_available_letters?` and `is_in_english_dict?` is Ruby convention that signals that it will return a boolean.

Good work!

Also, nice work on the optionals.

File: lib/adagrams.rb [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+      return false
+    end
+  end
+  return true
+end
+
+
+# wave 3
+
+def score_word (word)
```
Comment (tildeee) [inline lib/adagrams.rb:89]: just being nitpicky about syntax, but there technically shouldn't be a space between the method name and the opening parens for paramaters, aka: `score_word (word)` -> `score_word( word )`

File: lib/adagrams.rb [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+  end
+  return player_hand
+end
+
+#wave 2
+
+def uses_available_letters(input, letters_in_hand)
+
+#takes input word and converts into a array for each letter
+  input_split_array = input.scan(/\w/)
```
Comment (tildeee) [inline lib/adagrams.rb:72]: `.scan` ends up doing what you want it to do by grouping together every word character into an array... For clarity on the idea "takes input word and splits apart into an array for each letter," I may use the `.split('')` method instead, with splitting on each empty string (`''`). It's the same result, but maybe a different String method that may read more clearly
